<a href="https://colab.research.google.com/github/yagmurgcm/yagmurgecm/blob/main/hypothesis_tests.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# This study tests the following hypotheses:

**Interest Rate Changes and Search Trends:** Increases or decreases in TCMB’s interest rates correlate with fluctuations in search volumes for "faiz" (interest rate) and "dollar/euro."

**Inflation and Public Interest:**
Rising inflation leads to increased searches for "enflasyon" (inflation), as the public seeks more information about its impact on daily life.

# **Interest Rate Changes and Search Trends**

Null Hpothesis: There is no correlation between TCMB's interest rates and "interest rate" search volume.

Alternative Hypotesis: There is a positive correlation between interest rates and "interest rate" search volume.

If p-value is smaller than 0.05, reject Null Hypothesis.
********
Null Hpothesis: There is no correlation between TCMB's interest rates and "interest rate" search volume.

Alternative Hypotesis: There is a positive correlation between interest rates and "interest rate" search volume.

If p-value is smaller than 0.05, reject Null Hypothesis.
*********
Null Hpothesis: There is no correlation between TCMB's interest rates and "interest rate" search volume.

Alternative Hypotesis: There is a positive correlation between interest rates and "interest rate" search volume.

If p-value is smaller than 0.05, reject Null Hypothesis.


In [15]:
from scipy.stats import spearmanr
import pandas as pd

#find data
path = "/content/drive/MyDrive/DSA - HW/"
faiz_df = pd.read_csv(path + "faizTCMB.csv")
faiz_trend_df = pd.read_csv(path + "faiztrends.csv", skiprows=1)
dolar_df = pd.read_csv(path + "dolartrends.csv", skiprows=1)
euro_df = pd.read_csv(path + "eurotrends.csv", skiprows=1)

#group data
faiz_trend_df.columns = ["Week", "faiz_trend"]
faiz_trend_df["Week"] = pd.to_datetime(faiz_trend_df["Week"])
faiz_trend_df["Month"] = faiz_trend_df["Week"].dt.to_period("M")
monthly_faiz_trend = faiz_trend_df.groupby("Month")["faiz_trend"].mean().reset_index()

dolar_df.columns = ["Week", "dolar_trend"]
euro_df.columns = ["Week", "euro_trend"]
for df in [dolar_df, euro_df]:
    df["Week"] = pd.to_datetime(df["Week"])
    df["Month"] = df["Week"].dt.to_period("M")
monthly_dolar_trend = dolar_df.groupby("Month")["dolar_trend"].mean().reset_index()
monthly_euro_trend = euro_df.groupby("Month")["euro_trend"].mean().reset_index()

faiz_df["Date"] = pd.to_datetime(faiz_df["Date"])
faiz_df["Month"] = faiz_df["Date"].dt.to_period("M")
monthly_faiz = faiz_df[["Month", "Interest"]]

merged = monthly_faiz.merge(monthly_faiz_trend, on="Month")
merged = merged.merge(monthly_dolar_trend, on="Month")
merged = merged.merge(monthly_euro_trend, on="Month")

# Correlation analysis
faiz_corr = spearmanr(merged["Interest"], merged["faiz_trend"])
dolar_corr = spearmanr(merged["Interest"], merged["dolar_trend"])
euro_corr = spearmanr(merged["Interest"], merged["euro_trend"])

#Print
print("Interest vs interest rate = Correlation:", round(faiz_corr.correlation, 3), "--- p-value:", round(faiz_corr.pvalue, 5))
print("Interest vs dolar = Correlation:", round(dolar_corr.correlation, 3), "--- p-value:", round(dolar_corr.pvalue, 5))
print("Interest vs euro = Correlation:", round(euro_corr.correlation, 3), "--- p-value:", round(euro_corr.pvalue, 5))


Interest vs interest rate = Correlation: 0.623 --- p-value: 0.0
Interest vs dolar = Correlation: -0.428 --- p-value: 0.00065
Interest vs euro = Correlation: 0.225 --- p-value: 0.08324


**Interest vs interest rate:** There is a strong positive and statistically significant correlation between TCMB interest rates and searches for "faiz" (correlation = 0.623, p-value < 0.005), suggesting that as interest rates rise, public interest in the topic increases.

**Interest vs dolar:** A moderate negative and significant correlation is observed between interest rates and searches for "dolar" (correlation = –0.428, p-value ≈ 0.00065), indicating that higher rates may reduce public attention toward the dollar.

**Interest vs euro:** Correlation between interest rates and "euro" searches is weak and not statistically significant (correlation = 0.225, p-value ≈ 0.083), implying no strong or reliable relationship.

# **Inflation and Public Interest**

Null Hpothesis: There is no correlation between TCMB's inflation rates and "inflation" search volume.

Alternative Hypotesis: There is a positive correlation between inflation rates and "inflation" search volume.

If p-value is smaller than 0.05, reject Null Hypothesis.

In [14]:
import pandas as pd
from scipy.stats import spearmanr

#find data
path = "/content/drive/MyDrive/DSA - HW/"
inflation_df = pd.read_csv(path + "inflationTCMB.csv")
inflation_trend_df = pd.read_csv(path + "enflasyontrends.csv", skiprows=1)

#process data
inflation_trend_df.columns = ["Week", "enflasyon_trend"]
inflation_trend_df["Week"] = pd.to_datetime(inflation_trend_df["Week"])
inflation_trend_df["Month"] = inflation_trend_df["Week"].dt.to_period("M")
monthly_trend = inflation_trend_df.groupby("Month")["enflasyon_trend"].mean().reset_index()

#inflation data process
inflation_df["Date"] = pd.to_datetime(inflation_df["Date"])
inflation_df["Month"] = inflation_df["Date"].dt.to_period("M")
monthly_rate = inflation_df[["Month", "Annual Change (%)"]]

#merge them
merged = monthly_rate.merge(monthly_trend, on="Month", how="inner")
merged_cleaned = merged.dropna()

#Correlation test
result = spearmanr(merged_cleaned["Annual Change (%)"], merged_cleaned["enflasyon_trend"])

print("Inflation vs 'inflation'  = Correlation:", round(result.correlation, 3),
      "--- p-value:", format(result.pvalue, ".2e"))


Inflation vs 'inflation'  = Correlation: 0.674 --- p-value: 3.47e-09


There is a strong positive and statistically significant correlation between inflation rate and "enflasyon" search trends — meaning that as inflation rises, public interest in the topic also increases.